control-plane/csi-driver/src/context.rs

```diff
@@ -410,6 +420,25 @@ impl TryFrom<&HashMap<String, String>> for PublishParams {
     }
 }
 
+/// Validate the topology parameters.
+/// Errors out of nodeSpreadTopologyKey has same values as of
+/// node_has_topology_key/nodeAffinityTopologyLabel
+pub(crate) fn validate_topology_params(
+    map1: Option<HashMap<String, String>>,
+    map2: Option<HashMap<String, String>>,
+) -> Result<(), tonic::Status> {
+    if let (Some(map1), Some(map2)) = (map1, map2) {
+        let has_common_keys = map1.len() == map2.len() && map1.keys().all(|k| map2.contains_key(k));
+
+        if has_common_keys {
+            return Err(tonic::Status::invalid_argument(
+                "Invalid node topology. `nodeSpreadTopologyKey` can't have same values as `nodeHasTopologyKey` or `nodeAffinityTopologyLabel`",
```

Comment:
What about nodeHasTopologyKey and nodeAffinityTopologyLabel. Do they need validation between them as well?

Reply:
IMO, we dont need that , `nodeAffinityTopologyLabel` is actually a subset of `nodeHasTopologyKey`
So of both are given the with the same key then `nodeAffinityTopologyLabel` will take the presedence as it has key and value both